components/camel-djl/src/main/java/org/apache/camel/component/djl/model/CustomImageClassificationPredictor.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.djl.model;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.imageio.ImageIO;
+
+import ai.djl.Model;
+import ai.djl.inference.Predictor;
+import ai.djl.modality.Classifications;
+import ai.djl.translate.TranslateException;
+import ai.djl.translate.Translator;
+import org.apache.camel.Exchange;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomImageClassificationPredictor extends AbstractPredictor {
+    private static final Logger LOG = LoggerFactory.getLogger(CustomImageClassificationPredictor.class);
+
+    private String modelName;
+    private String translatorName;
+
+    public CustomImageClassificationPredictor(String modelName, String translatorName) {
+        this.modelName = modelName;
+        this.translatorName = translatorName;
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        Model model = exchange.getContext().getRegistry().lookupByNameAndType(modelName, Model.class);
+        Translator translator = exchange.getContext().getRegistry().lookupByNameAndType(translatorName, Translator.class);
+
+        if (exchange.getIn().getBody() instanceof byte[]) {
+            byte[] bytes = exchange.getIn().getBody(byte[].class);
+            Map<String, Float> result = classify(model, translator, new ByteArrayInputStream(bytes));
+            exchange.getIn().setBody(result);
+        } else if (exchange.getIn().getBody() instanceof File) {
+            Map<String, Float> result = classify(model, translator, exchange.getIn().getBody(File.class));
+            exchange.getIn().setBody(result);
+        } else if (exchange.getIn().getBody() instanceof InputStream) {
+            Map<String, Float> result = classify(model, translator, exchange.getIn().getBody(InputStream.class));
+            exchange.getIn().setBody(result);
+        }
```

Comment:
Does it suppose to throw an error in case non of these conditions being satisfied?

Reply:
Fixed for all classes that extends AbstractPredictor